contracts/storage/ProfileStorage.sol

```diff
@@ -49,7 +54,66 @@ contract ProfileStorage {
         _;
     }
 
+    function createProfile(
+        address operationalWallet,
+        address managementWallet,
+        bytes nodeId,
+        uint96 initialAsk,
+        uint96 initialStake
+    )
+        public
+        OnlyContracts
+        returns (uint96, address)
+    {
+        require(!identityIds[operationalWallet] && !identityIds[managementWallet], "Profile already exists");
```

Comment:
Then I guess it makes sense not to add management wallets to the mapping with identityIds at all, because it will require bigger changes (one-many relation). Is it a proper way to go if we only can get identityId/identityContractAdress, using operational wallet?

Reply:
actually we should have identityIds map to identityContractAddress only, and then check if msg.sender is a key in identityContract with the right purpose. 
I would remove this mapping entirely actually.

Reply:
But where do we get IdentityId from without this mapping?